File: src/main/java/kr/hahaha98757/zombiesaddon/commands/CommandWaveDelays.java
package kr.hahaha98757.zombiesaddon.commands;

import kr.hahaha98757.zombiesaddon.enums.Difficulty;
import kr.hahaha98757.zombiesaddon.features.WaveDelays;
import kr.hahaha98757.zombiesaddon.utils.Utils;
import net.minecraft.client.Minecraft;
import net.minecraft.command.CommandBase;
import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;
import net.minecraft.command.WrongUsageException;
import net.minecraft.util.BlockPos;

import java.util.List;

public class CommandWaveDelays extends CommandBase {

    @Override
    public String getCommandName() {
        return "wavedelays";
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 0;
    }

    @Override
    public String getCommandUsage(ICommandSender sender) {
        return "/wavedelays difficult <normal|hard|rip>";
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) throws CommandException {
        if (sender != Minecraft.getMinecraft().thePlayer) return;

        if (args.length < 2) throw new WrongUsageException(getCommandUsage(null));

        switch (args[1]) {
            case "normal":
                WaveDelays.setDifficulty(Difficulty.NORMAL);
                Utils.addTranslationChat("zombiesaddon.commands.wavedelays.success", "§aNormal");
                break;
            case "hard":
                WaveDelays.setDifficulty(Difficulty.HARD);
                Utils.addTranslationChat("zombiesaddon.commands.wavedelays.success", "§cHard");
                break;
            case "rip":
                WaveDelays.setDifficulty(Difficulty.RIP);
                Utils.addTranslationChat("zombiesaddon.commands.wavedelays.success", "§4RIP");
                break;
            default:
                throw new WrongUsageException(getCommandUsage(null));
        }
    }

    @Override
    public List<String> addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos) {
        if (args.length == 1) return getListOfStringsMatchingLastWord(args, "difficult");
        else if (args.length == 2) return getListOfStringsMatchingLastWord(args, "normal", "hard", "rip");
        return null;
    }
}

File: src/main/kotlin/kr/hahaha98757/zombiesaddon/commands/CommandWaveDelays.kt
package kr.hahaha98757.zombiesaddon.commands

import kr.hahaha98757.zombiesaddon.ZombiesAddon
import kr.hahaha98757.zombiesaddon.enums.Difficulty
import kr.hahaha98757.zombiesaddon.modules.WaveDelays
import kr.hahaha98757.zombiesaddon.utils.addTranslationChat
import net.minecraft.command.ICommandSender
import net.minecraft.command.WrongUsageException
import net.minecraft.util.BlockPos

class CommandWaveDelays: CustomCommandBase() {
    override fun getCommandName() = "wavedelays"
    override fun getCommandUsage(sender: ICommandSender?) = "/wavedelays <difficulty|offset> <normal|hard|rip>"
    override fun runCommand(sender: ICommandSender, args: Array<String>) {
        if (args.isEmpty() || args.size < 2) throw WrongUsageException(getCommandUsage(null))

        when (args[0]) {
            "difficulty" -> {
                val difficulty = when (args[1].lowercase()) {
                    "normal" -> {
                        WaveDelays.instance.difficulty = Difficulty.NORMAL
                        "§aNormal"
                    }
                    "hard" -> {
                        WaveDelays.instance.difficulty = Difficulty.HARD
                        "§cHard"
                    }
                    "rip" -> {
                        WaveDelays.instance.difficulty = Difficulty.RIP
                        "§4RIP"
                    }
                    else -> throw WrongUsageException(getCommandUsage(null))
                }
                addTranslationChat("zombiesaddon.commands.wavedelays.difficulty", difficulty)
            }
            "offset" -> {
                val offset = parseInt(args[1], -1000, 1000)
                ZombiesAddon.instance.config.waveDelaysRLModeOffset = offset
                addTranslationChat("zombiesaddon.commands.wavedelays.offset", "§a$offset")
            }
        }
    }

    override fun addTabCompletionOptions(sender: ICommandSender?, args: Array<String?>, pos: BlockPos?): List<String>? {
        if (args.isEmpty()) return null
        if (args.size == 1) return getListOfStringsMatchingLastWord(args, "difficulty", "offset")
        if (args.size != 2) return null
        return when (args[0]) {
            "difficulty" -> getListOfStringsMatchingLastWord(args, "normal", "hard", "rip")
            "offset" -> getListOfStringsMatchingLastWord(args, "-28")
            else -> null
        }
    }
}